docs/faq.rst

```diff
@@ -25,15 +25,37 @@ First of all: Please (please, please) do NOT run Rally against your production c
-
-The explanation of `latency` is a bit more involved. First of all, Rally defines two benchmarking modes:
-
-* Throughput benchmarking mode: In this mode, Rally will issue requests as fast as it can, i.e. as soon as it receives a response, it will issue the next request. This is ideal for benchmarking indexing. In this mode ``latency`` == ``service_time``.
-* Throughput-throttled mode: If you define a specific target throughput rate in a track, for example 100 requests per second (you should choose this number based on the traffic pattern that you experience in your production environment), then Rally will define a schedule internally and will issue requests according to this schedule regardless how fast Elasticsearch can respond. To put it differently: Imagine you want to grab a coffee on your way to work. You make this decision independently of all the other people going to the coffee shop so it is possible that you need to wait before you can tell the barista which coffee you want. The time it takes the barista to make your coffee is the service time. The service time is independent of the number of customers in the coffee shop. However, you as a customer also care about the length of the waiting line which depends on the number of customers in the coffee shop. The time it takes between you entering the coffee shop and taking your first sip of coffee is latency.
+What do `latency` and `service_time` mean and how do they relate to the `took` field that Elasticsearch returns?
+----------------------------------------------------------------------------------------------------------------
+
+The ``took`` field is included by Elasticsearch in responses to searches and
+bulk indexing and reports the time that Elasticsearch spent processing the
+request. This value is measured on the server so it includes neither the time
+it took for the request to get from the client to Elasticsearch nor the time it
+took for the response to arrive back at the client. In contrast,
+``service_time`` is measured by Rally as the length of time from when it
+started to send the request to Elasticsearch until it finishing receiving the
+response. Therefore ``service_time`` does include the time spent sending the
+request and receiving the response.
+
+The explanation of ``latency`` depends on your choice of benchmarking mode:
+
+* **Throughput benchmarking mode**: In this mode, Rally will issue requests as
+  fast as it can: as soon as it receives a response to one request it will
+  generate and send the next request. In this mode ``latency`` and
+  ``service_time`` are identical.
+
+* **Throughput-throttled mode**: You may prefer to run benchmarks with a target
```

Comment:
Internally Rally uses a concept called "Schedule". There are different ways to alter the traffic that Rally generates by issuing requests:

- target-throughput (in requests/sec)
- target-interval (inverse of target-throughput i.e. amount of secs to wait per request)
- use poisson distribution or a custom schedule of your own

As all of those actually end up influencing the schedule that Rally will use to issue requests, I wonder if we should be talking about the schedule here briefly and adjust wording below, e.g. in line 51/52 instead of "according to this target" "according to the schedule".

However ... I think this concept is tricky by itself and by trying to provide a very generic answer (as I suggest) we might be confusing the reader, whereas the description in this PR is easier to understand by using a concrete example (via target throughput). @danielmitterdorfer WDYT?

Reply:
Ack, thanks for the clarification. I'll try and generify the wording and maybe link to the schedule docs if I can work out how.

Reply:
I pushed some more generic words in https://github.com/elastic/rally/pull/1133/commits/18ae7f27e4f191b153c184e74fdef5b7cdaf6a45 which I don't think make it any more complicated.

I tried to work out how to cross-reference the docs on the `schedule` bit of a track but timed out after a couple of minutes of trying to understand what on earth the [Sphinx docs](https://www.sphinx-doc.org/en/1.7/markup/inline.html) were trying to tell me. A cross-ref on the first occurrence of the word `schedule` would be very helpful IMO, could someone else push an update that adds one?

Reply:
> could someone else push an update that adds one?

@DaveCTurner I've added one; it's available if you merge the latest master into your branch. You can use it as follows:

```
:ref:`your custom link text here <track_choose_schedule>`
```

Reply:
Thanks @danielmitterdorfer, done in 4f999a72de81c2f703ed41b8ce9151d93a58a6fd.